### DISCUSSION TITLE
cannot move images to empty folder

### DISCUSSION THREAD

**Discussion by Raoul-Kima** [deckerst/aves#632] [Ideas]
I'm unsure if this is a bug report, feature request or user interface discussion.

It's difficult to move specific images from the smartphone to the PC, because because computers don't show thumbnails for images on the smartphone (or at least mine doesnt, on recent android versions). To circumvent this i like to first use a smartphone app to move the images to an otherwise empty folder, and then simply grab everything in there from the computer.
Problem: I cannot move images to an empty folder with aves, because empty folders are not shown and I also cannot use the "create new folder" feature because that one (correctly) complains that a folder with that name already exists.
I do want to use the gallery app for this rather then the file manager, because the file manager doest have such good thumbnails, making it more difficult to see which files I want to transfer.

The easy workaround: Just create a new folder every time.

The (maybe?) optimal solution: Have a file manager with good thumbnails and mosaicing and such instead of separate file-managing and image-browsing apps.

But maybe it's also worth considering to add a way to somehow add images to empty folders with the image browsing app. Maybe it would make sense to make it so the app always shows all folders inside "pictures", even if empty? The ability to have empty "collections" seems like a basic image management feature? But I understand that this might be considered to be outside the scope of this app, since its not a file manager.

**Comment by deckerst** [comment]
> computers don't show thumbnails for images on the smartphone

Just for your information, I use Ubuntu (vanilla), and it displays thumbnails for images on the Samsung mobile I connect to it.

> Problem: I cannot move images to an empty folder with aves, because empty folders are not shown and I also cannot use the "create new folder" feature because that one (correctly) complains that a folder with that name already exists.

I understand from this sentence that it is important to you to move these images to the same directory (at the same path) every time. Is it because you use a tool to automate moving from that fixed directory to your computer? Otherwise, you can just give it a unique temporary name on the spot.

> Maybe it would make sense to make it so the app always shows all folders inside "pictures", even if empty?

Not going to do that, but one thing I could do is to keep empty directories while they're pinned. So if you don't want to have a directory disappear when it's getting empty, just pin it. It wouldn't be tied to a specific location (like what you suggested with `all folders inside "pictures"`).

Also, maybe I should let the app create an empty album even if the name matches an existing directory. That would also solve you problem but you would have to recreate the album every time.

**Comment by Raoul-Kima** [comment]
> I use Ubuntu (vanilla), and it displays thumbnails

Is your mobile connecting as a usb-mass-drive or via media transfer protocoll or similar? I only get thumbnails with mass-storage (which newer android devices seem to typically not support).

> I understand from this sentence that it is important to you to move these images to the same directory (at the same path) every time. Is it because you use a tool to automate moving from that fixed directory to your computer? Otherwise, you can just give it a unique temporary name on the spot.

No, I'm moving it manually. I can give the folder a new temporary name. It's just less fiddely when the name is always the same and I dont have to delete/remake the folder every time. It's not a big deal, just a detail optimisation.

> one thing I could do is to keep empty directories while they're pinned

Sounds like a good solution for this problem.

**Reply by deckerst** [reply]
> Is your mobile connecting as a usb-mass-drive or via media transfer protocoll or similar? I only get thumbnails with mass-storage (which newer android devices seem to typically not support).

```bash
tibo@redstart:~$ lsusb
...
Bus 001 Device 008: ID 04e8:6860 Samsung Electronics Co., Ltd Galaxy A5 (MTP)
...
```
So from Ubuntu's point of view it's MTP.

And here's the USB config on the phone (Samsung S10e):
<img width="250" src="https://github.com/deckerst/aves/assets/13898333/648e240e-cda9-40fe-b926-f394f837ffdf" />

**Comment by Raoul-Kima** [comment]
> it's MTP
Hm, ok thanks!
So there is a way to show thumbnails through MTP.
